src/arithmetic/arithmetic_expression.c

```diff
@@ -421,8 +428,14 @@ static AR_EXP_Result _AR_EXP_EvaluateFunctionCall
 	int child_count = node->op.child_count;
 
 	// evaluate each child before evaluating current node
-	SIValue sub_trees[child_count];
-
+    SIValue *sub_trees;
+    // if array size is above the threshold, we allocate it on the heap (otherwise on stack)
+    SIValue sub_trees_on_stack[MAX_ARRAY_SIZE_ON_STACK];
```

Comment:
why always allocate on the stack?
and why always the max num?

Reply:
If I allocate on a stack inside of a block (such as `if` statement) , the variable is no longer valid when we are out of scope.
But after consulting with Roi, I will add a change that will save the unnecessary allocation on the stack.